### PR TITLE
Contact restrictions: Errors in the UI

### PR DESCRIPTION
**Pull request by cjb**
@keybase/picnicsquad CC @keybase/react-hackers 

Here's the GUI side of being on the losing end of contact restrictions.  The PR:

* fixes the "createConversationError" integration -- we used to have a screen for when creating a new convo fails, but it lost its reducer and just black bars on master.  Unlike the other cases, this one isn't a modal, it appears in place of the conversation itself.
* adds try/catch around KBFS new private folder creation, shows the modal if it failed because of contact restriction
* adds try/catch around requesting payment, shows the modal if it failed because of contact restriction
* adds try/catch around adding people to a team.  There are two possible failure modes: if some people could be added, the RPC succeeds and sends us the list of failures, and everyone else was added. If no-one could be added, the RPC errors out.
* If you're trying to add people to a conversation, and some of them are blocked but some aren't, the error page has a "Create without them" button that retries the conversation with the blocked members missing.
* creates a new component (`team-building/contact-restricted.tsx`) which is used for these cases as a modal.

Some screens:

Screen Shot 2020-01-09 at 11 49 32 AM
Screen Shot 2020-01-09 at 11 49 51 AM
Screen Shot 2020-01-09 at 11 51 56 AM

### CODE REVIEW

**Comment by jakob223** [comment]
Is this ready for review?

**Comment by cjb** [comment]
@jakob223 yes!

**Comment by jakob223** [comment]
I can get a black bar by trying to add someone on contact restrictions with the chat button on the people page:

image

File: shared/team-building/contact-restricted.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  switch (props.source) {
+    case 'walletsRequest':
+      header = `You cannot request a payment from @${props.usernames[0]}.`
+      description = `@${props.usernames[0]}'s contact restrictions prevent you from requesting a payment. Contact them outside Keybase to proceed.`
+      break
+    case 'newFolder':
+      header = `You cannot open a private folder with @${props.usernames[0]}.`
+      description = `@${props.usernames[0]}'s contact restrictions prevent you from opening a private folder with them. Contact them outside Keybase to proceed.`
+      break
+    case 'teamAddAllFailed':
```
Comment (jakob223) [inline shared/team-building/contact-restricted.tsx]: Can we add a special case for when there's only one user here?

File: shared/teams/routes.tsx [hunk 2]
```diff
@@ -23,6 +24,9 @@ export const newRoutes = {
 
 export const newModalRoutes = {
   // TODO connect broken
```
Comment (jakob223) [inline shared/teams/routes.tsx]: probably this comment wants to stay with `retentionWarning`

**Comment by cjb** [comment]
@jakob223 Cool, thanks for the review! Made these changes.

**Pull request merged**